lib/string_decoder.js

```diff
@@ -51,6 +51,12 @@ const kNativeDecoder = Symbol('kNativeDecoder');
 
 // Do not cache `Buffer.isEncoding` when checking encoding names as some
 // modules monkey-patch it to support additional encodings
+/**
+ *
+ * @param {string} enc
+ * @returns {"utf8" | "utf16le" | "hex" | "ascii"
+ *           | "base64" | "latin1" | "base64url"}
```

Comment:
This function also throws an error.

```suggestion
 *           | "base64" | "latin1" | "base64url"}
 * @throws {TypeError} Throws an error when encoding is Invalid
```